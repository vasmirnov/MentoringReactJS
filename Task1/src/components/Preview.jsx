import React from 'react';
import * as s from './preview.css';
export class Preview extends React.Component {
    render() {
        return (
            <div className={s.main}>
                <img className={s.img} src={this.props.img}/>
                <div className={s.title}>{this.props.title}</div>
                <div className={s.year}>{this.props.year}</div>
                <div className={s.genre}>{this.props.genre}</div>
            </div>
        )
    }
}