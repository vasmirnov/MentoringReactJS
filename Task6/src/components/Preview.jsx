import React from 'react';
import * as s from './preview.css';
import defaultImg from '../img/default.jpg';


export class Preview extends React.PureComponent {
    constructor(...args) {
        super(...args);

        this.selectFilmHandler = this.selectFilmHandler.bind(this);
    }
    selectFilmHandler(film) {
        this.props.selectFilmHandler(this.props.film.get("id"));
    }

    getImageLing(film) {
        var img = film.get("img");
        return img && "https://image.tmdb.org/t/p/w300" + film.get("img") || defaultImg;
    }
    render() {
        return (
            <div className={s.main} onClick={this.selectFilmHandler}>
                <img className={s.img} src={this.getImageLing(this.props.film)} />
                <div className={s.title}>{this.props.film.get("title")}</div>
                <div className={s.year}>{this.props.film.get("year")}</div>
                <div className={s.genre}>{this.props.film.get("genre")}</div>
            </div>
        )
    }
}