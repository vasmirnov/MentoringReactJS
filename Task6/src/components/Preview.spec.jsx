import React from 'react';
import renderer from 'react-test-renderer';
import { Preview } from './Preview';
import { Map } from 'immutable';

jest.disableAutomock();
jest.mock('../img/default.jpg', () => 'defaultImg');

jest.mock('./preview.css', () => {
    return {
        main: 'main',
        img: 'img',
        title: 'title',
        year: 'year',
        genre: 'genre'
    }
});

let props;

describe('Preview', () => {

    test('render general', () => {
        const props = {
            selectFilmHandler: () => { },
            film: Map({
                id: 1,
                img: '1.jpg',
                title: 'Kill Bill 1',
                year: '2001',
                genre: 'comedy'
            })
        }
        const instance = renderer.create(
            <Preview {...props} />
        );
        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('render default img', () => {
        const props = {
            selectFilmHandler: () => { },
            film: Map({
                id: 1,
                title: 'Kill Bill 2',
                year: '2003',
                genre: 'comedy'
            })
        }
        const instance = renderer.create(
            <Preview {...props} />
        );
        expect(instance.toJSON()).toMatchSnapshot();
    });
});
