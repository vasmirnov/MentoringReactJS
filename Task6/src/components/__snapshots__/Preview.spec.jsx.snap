// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview render default img 1`] = `
<div
  className="main"
  onClick={[Function]}
>
  <img
    className="img"
    src="defaultImg"
  />
  <div
    className="title"
  >
    Kill Bill 2
  </div>
  <div
    className="year"
  >
    2003
  </div>
  <div
    className="genre"
  >
    comedy
  </div>
</div>
`;

exports[`Preview render general 1`] = `
<div
  className="main"
  onClick={[Function]}
>
  <img
    className="img"
    src="https://image.tmdb.org/t/p/w3001.jpg"
  />
  <div
    className="title"
  >
    Kill Bill 1
  </div>
  <div
    className="year"
  >
    2001
  </div>
  <div
    className="genre"
  >
    comedy
  </div>
</div>
`;
